File: YJKit-ObjC/YJKit-ObjC/YJKit/YJKUtil.m
//
//  YJKUtil.m
//  YJKit-ObjC
//
//  Created by 杨坚 on 2017/5/16.
//  Copyright © 2017年 杨坚. All rights reserved.
//

#import "YJKUtil.h"

NSString * cachesPath(void)
{
    static NSString *path = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        path = NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES)[0];
    });
    return path;
}

BOOL simpleChinese(void)
{
    static BOOL value = NO;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        NSArray *allLanguages = [[NSUserDefaults standardUserDefaults] objectForKey:@"AppleLanguages"];
        NSString *preferredLang = allLanguages[0];
        value = [preferredLang hasPrefix:@"zh-Hans"];
    });
    return value;
}

BOOL traditionalChinese(void)
{
    static BOOL value = NO;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        NSArray *allLanguages = [[NSUserDefaults standardUserDefaults] objectForKey:@"AppleLanguages"];
        NSString *preferredLang = allLanguages[0];
        value = [preferredLang hasPrefix:@"zh-Hant"];
    });
    return value;
}

File: YJKit-Swift/YJKit-Swift/YJKit/YJKUtil.swift
//
//  YJKUtil.swift
//  YJKit-Swift
//
//  Created by 杨坚 on 2017/5/17.
//  Copyright © 2017年 杨坚. All rights reserved.
//

import Foundation

/// 获取caches目录
public func cachesPath() -> String {
    return NSSearchPathForDirectoriesInDomains(.cachesDirectory, .userDomainMask, true)[0]
}

/// 是否简体中文
public func simpleChinese() -> Bool {
    return (UserDefaults.standard.stringArray(forKey: "AppleLanguages")?[0].hasPrefix("zh-Hans"))!
}

/// 是否繁体中文
public func traditionalChinese() -> Bool {
    return (UserDefaults.standard.stringArray(forKey: "AppleLanguages")?[0].hasPrefix("zh-Hant"))!
}
